test-integration/filter-ens.js

```diff
@@ -0,0 +1,189 @@
+/* global artifacts */
+
+const ethers = require("ethers");
+const truffleAssert = require("truffle-assertions");
+const ArgentContext = require("../utils/argent-context.js");
+const utils = require("../utils/utilities.js");
+
+const ENSRegistry = artifacts.require("ENSRegistry");
+const ArgentENSManager = artifacts.require("ArgentENSManager");
+const ArgentENSResolver = artifacts.require("ArgentENSResolver");
+const ReverseRegistrar = artifacts.require("ReverseRegistrar");
+const Filter = artifacts.require("TestFilter");
+
+const ARGENT_ENS_ADDRESS = "0xF32FDDEF964b98b1d2d2b1C071ac60ED55d4D217";
+const ARGENT_ENS_OWNER_ADDRESS = "0xa5c603e1C27a96171487aea0649b01c56248d2e8";
+
+const { ZERO_ADDRESS } = utils;
+
+contract("ENS contracts", (accounts) => {
+  let infrastructure;
+  let owner;
+  const amanager = accounts[3];
+  const anonmanager = accounts[5];
+  const recipient = accounts[8];
+
+  let argent;
+  const root = "xyz";
+  const subnameWallet = "argent";
+  const walletNode = ethers.utils.namehash(`${subnameWallet}.${root}`);
+  const uniqueLabel = () => Math.random().toString().slice(2);
+
+  let ensRegistry;
+  let ensResolver;
+  let ensReverse;
+  let argentEnsManager;
+
+  before(async () => {
+    argent = await new ArgentContext(accounts).initialise();
+    ({ infrastructure, owner } = argent);
+
+    argentEnsManager = await ArgentENSManager.at(ARGENT_ENS_ADDRESS);
+    await argentEnsManager.addManager(infrastructure, { from: ARGENT_ENS_OWNER_ADDRESS });
+
+    ensRegistry = await ENSRegistry.at(await argentEnsManager.ensRegistry());
+    ensResolver = await ArgentENSResolver.at(await argentEnsManager.ensResolver());
+    ensReverse = await ReverseRegistrar.at(await argentEnsManager.getENSReverseRegistrar());
+
+    const filter = await Filter.new();
+    await argent.dappRegistry.addDapp(0, argentEnsManager.address, filter.address);
+    await argent.dappRegistry.addDapp(0, recipient, ZERO_ADDRESS);
+  });
+
+  describe("ENS Manager", () => {
```

Comment:
I have the impression that all these tests are unit tests of the ArgentENSManager which should stay in `/test/ens.js` as they are today. Here we should only test the integration with a wallet (i.e. the calls initiated by a wallet and that go through the filter).

Reply:
Makes sense, this one doesn't need to be duplicated